cc2538-bsl.py

```diff
@@ -784,11 +784,12 @@ def read(filename):
                 length = length + (length % 4)
```

Comment:
You're trying to round up to a multiple of 4, yes? How about `(length + 3) & ~0x03`?